transform/app/transform/transformers/catalogue.py

```diff
@@ -0,0 +1,103 @@
+# pylint: disable=line-too-long, wildcard-import, unused-wildcard-import, invalid-name, duplicate-code
+"""Transform Marketplace's resources"""
+from pyspark.sql import SparkSession, DataFrame
+from pyspark.sql.functions import lit, col
+from pyspark.sql.types import StringType, StructType
+from app.transform.transformers.base.base import BaseTransformer
+from app.transform.schemas.properties.data import (
+    ID,
+    CREATED_AT,
+    KEYWORDS,
+    NAME,
+    PUBLICATION_DATE,
+    TAG_LIST,
+    TITLE,
+    TYPE,
+)
+from app.transform.schemas.output.catalogue import catalogue_output_schema
+from app.settings import settings
+
+
+class CatalogueTransformer(BaseTransformer):
+    """Transformer used to transform catalogues"""
+
+    def __init__(self, spark: SparkSession):
+        self.type = settings.CATALOGUE
+        # Increase the range of providers IDs -> to avoid a conflicts
+        self.id_increment = settings.CATALOGUE_IDS_INCREMENTOR
+        self.exp_output_schema = catalogue_output_schema
+
+        super().__init__(
+            self.type,
+            self.cols_to_add,
+            self.cols_to_drop,
+            self.cols_to_rename,
+            self.exp_output_schema,
+            spark,
+        )
+
+    def apply_simple_trans(self, df: DataFrame) -> DataFrame:
+        """Apply simple transformations.
+        Simple in a way that there is a possibility to manipulate the main dataframe
+        without a need to create another dataframe and merging"""
+        df = df.withColumn(TYPE, lit(self.type))
+        df = self.rename_cols(df)
+        df = df.withColumn(ID, (col(ID) + self.id_increment))
+
+        return df
+
+    def apply_complex_trans(self, df: DataFrame) -> DataFrame:
+        """Harvest oag properties that requires more complex transformations
+        Basically from those harvested properties there will be created another dataframe
+        which will be later on merged with the main dataframe"""
+
+        raise df
+
+    @staticmethod
+    def cast_columns(df: DataFrame) -> DataFrame:
+        """Cast certain columns"""
+        df = (
+            df.withColumn("id", col("id").cast(StringType()))
+            # .withColumn("created_at", col("updated_at").cast("date"))
```

Comment:
why it's commented out?

Reply:
My bad, I forgot about it. changed into: `.withColumn("created_at", col("created_at").cast("date"))`